### PR TITLE
From<::std::io::Error> for Error and change version error-chain to 0.12

### PR DESCRIPTION
**Pull request by P0ntiuS**
Fix amq-proto compile problem 
amq-proto-compile-error

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -1,3 +1,10 @@
+#[cfg(windows)]
+impl From<::std::io::Error> for Error {
```
Comment (andir) [inline src/error.rs]: Wouldn't it be better to remove/extend the conditional from the 
`foreign_links` section below?

**Review by andir — CHANGES_REQUESTED** [review]

**Comment by Riateche** [comment]
This fix is working. It would be nice if this was merged and the new version was published. As it is now, the windows build is completely broken.

File: src/error.rs (hunk 1 above)
Comment (P0ntiuS) [reply src/error.rs]: done

**Comment by P0ntiuS** [comment]
> 
> 
> This fix is working. It would be nice if this was merged and the new version was published. As it is now, the windows build is completely broken.

I think now he should take it.

**Pull request closed without merging**